### PR TITLE
Plant hydraulics restart

### PR DESCRIPTION
**Pull request by pnlfang**
Restart from existing land files that do not contain required variables
for plant hydraulics.

[BFB]

### CODE REVIEW

File: components/clm/src/biogeophys/SoilStateType.F90 [hunk 1]
```diff
@@ -881,11 +881,13 @@ subroutine Restart(this, bounds, ncid, flag)
             dim1name='column', dim2name='levgrnd', switchdim=.true., &
             long_name='soil matric potential', units='mm', &
             interpinic_flag='interp', readvar=readvar, data=this%smp_l_col)
+       if(flag=='read' .and. .not.readvar) this%smp_l_col = -2.5e3_r8
```
Comment (rljacob) [inline components/clm/src/biogeophys/SoilStateType.F90:884]: What is this number?   Add a comment explaining why you are setting this variable to that number.

File: components/clm/src/biogeophys/SoilStateType.F90 [hunk 2]
```diff
@@ -881,11 +881,13 @@ subroutine Restart(this, bounds, ncid, flag)
...
 
        call restartvar(ncid=ncid, flag=flag, varname='HK', xtype=ncd_double,  &
             dim1name='column', dim2name='levgrnd', switchdim=.true., &
             long_name='hydraulic conductivity', units='mm/s', &
             interpinic_flag='interp', readvar=readvar, data=this%hk_l_col)
+       if(flag=='read' .and. .not.readvar) this%hk_l_col = 1.e-9_r8
```
Comment (rljacob) [inline components/clm/src/biogeophys/SoilStateType.F90:890]: Same comment as above.  What is the number and why are you assigning it to that variable here?

Comment (pnlfang) [reply components/clm/src/biogeophys/SoilStateType.F90:894]: Added comments. Thanks!

**Comment by rljacob** [comment]
@jqyin are you able to merge this?

**Comment by rljacob** [comment]
@bishtgautam I re-assigned this to you.

**Comment by bishtgautam** [comment]
@pnlfang The current PR title is too long, please rename it. Also, give a description of this PR. Details about submitting PR is available [here](https://acme-climate.atlassian.net/wiki/spaces/ED/pages/2523163/Development+Reference#DevelopmentReference-Submittingapullrequest(PR)). 

In the future, please follow the E3SM branch naming convention that is described [here](https://acme-climate.atlassian.net/wiki/spaces/ED/pages/2523172/Branch+Tag+and+Version+name+conventions) (e.g. `pnlfang/lnd/plant-hydraulics-restart`).

**Comment by bishtgautam** [comment]
@pnlfang Any update on addressing my [earlier comment](https://github.com/E3SM-Project/E3SM/pull/3755#issuecomment-687426822)?

**Comment by pnlfang** [comment]
This change will allow restart from existing land files that do not contain required variables for plant hydraulics simulation.

**Comment by vanroekel** [comment]
@bishtgautam yes, I approve this going to master assuming @golaz agrees.

**Comment by bishtgautam** [comment]
@golaz Is it okay to merge this PR to master?

cc: @rljacob

**Comment by golaz** [comment]
Thanks for the reminder. I thought this had already been merged in.

@rljacob, we can include this in the v2 code base as long as it it is BFB with all current Water Cycle compsets (CMIP6 style). There will be a separate decision later whether we turn on plant hydraulics or not for v2.

**Pull request merged**